scripts/prebuild/run-prebuild.mjs

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { execFileSync } from 'node:child_process'
+import { platform, arch } from 'node:os'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+import {
+	existsSync,
+	writeFileSync,
+	chmodSync,
+	readFileSync,
+	unlinkSync,
+} from 'node:fs'
+import { gunzipSync } from 'node:zlib'
+
+function unGzip(filename) {
+	const compressedFile = `${filename}.gz`
+	const compressedData = readFileSync(compressedFile)
+	const decompressedData = gunzipSync(compressedData)
+	writeFileSync(filename, decompressedData)
+}
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url))
+
+/**
+ * Detect system architecture and run the appropriate prebuild program
+ */
+async function runPrebuild() {
+	const systemPlatform = platform()
+	const systemArch = arch()
+
+	console.log(
+		`Detected platform: ${systemPlatform}, architecture: ${systemArch}`,
+	)
+
+	let filename
+	let binaryExists = false
+
+	if (systemPlatform === 'darwin' && systemArch === 'arm64') {
+		binaryExists = true
+		filename = `${path.join(__dirname, 'prebuild-arm-mac-binary')}`
+		console.log('Running prebuild binary for ARM Mac...')
+	} else if (systemPlatform === 'linux' && systemArch === 'arm64') {
+		binaryExists = true
+		filename = `${path.join(__dirname, 'prebuild-arm-linux-binary')}`
+		console.log('Running prebuild binary for ARM Linux...')
+	} else if (systemPlatform === 'linux' && systemArch === 'x64') {
+		binaryExists = true
+		filename = `${path.join(__dirname, 'prebuild-x64-linux-binary')}`
+		console.log('Running prebuild binary for x64 Linux...')
+	} else {
+		// Default fallback
+		console.log('Running default node prebuild...')
+	}
+
+	if (binaryExists) {
+		try {
+			if (existsSync(filename)) {
```

Comment:
Yup we delete the old binary and unzip a new binary each time. It's super duper quick and it means that we **don't** have to worry about folks having old binaries running. Which is a big win.